oxia-operator/controllers/oxiacluster_coordinator.go

```diff
@@ -0,0 +1,218 @@
+		if !apiErrors.IsNotFound(err) {
+			r.Log.Error(err, "Failed to get role binding: ", "NamespaceName", nsName)
+			return ctrl.Result{}, err
+		}
+		r.Log.Info("Creating a new role binding:", "NamespaceName", nsName)
+		err := r.Client.Create(ctx, roleBinding(Coordinator, oxia))
+		if err != nil {
+			r.Log.Error(err, "Failed to create role binding:", "NamespaceName", nsName)
+			return ctrl.Result{}, err
+		}
+		r.Log.Info("Created role binding: ", "NamespaceName", nsName)
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if oxia.DeletionTimestamp != nil {
+		r.Log.Info("Role binding deletion is unsupported yet!")
+	}
+
+	// reconcile configmap
+	_configMap := &coreV1.ConfigMap{}
+	nsName = types.NamespacedName{Name: resourceName(Coordinator, oxia.Name), Namespace: oxia.Namespace}
+	err = r.Client.Get(ctx, nsName, _configMap)
+	if err != nil {
+		if !apiErrors.IsNotFound(err) {
+			r.Log.Error(err, "Failed to get configmap: ", "NamespaceName", nsName)
+			return ctrl.Result{}, err
+		}
+		r.Log.Info("Creating a new configmap:", "NamespaceName", nsName)
+		configMap, err := coordinatorConfigMap(oxia)
+		if err != nil {
+			r.Log.Error(err, "Failed to create configmap:", "NamespaceName", nsName)
+			return ctrl.Result{}, err
+		}
+		err = r.Client.Create(ctx, configMap)
+		if err != nil {
+			r.Log.Error(err, "Failed to create configmap:", "NamespaceName", nsName)
+			return ctrl.Result{}, err
+		}
+		r.Log.Info("Created configmap: ", "NamespaceName", nsName)
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if oxia.DeletionTimestamp != nil {
+		r.Log.Info("Configmap deletion is unsupported yet!")
+	}
+
+	// reconcile deployment
+	_deployment := &appsV1.Deployment{}
+	nsName = types.NamespacedName{Name: resourceName(Coordinator, oxia.Name), Namespace: oxia.Namespace}
+	err = r.Client.Get(ctx, nsName, _deployment)
+	if err != nil {
+		if !apiErrors.IsNotFound(err) {
+			r.Log.Error(err, "Failed to get deployment: ", "NamespaceName", nsName)
+			return ctrl.Result{}, err
+		}
+		r.Log.Info("Creating a new deployment:", "NamespaceName", nsName)
+		err = r.Client.Create(ctx, coordinatorDeployment(oxia))
+		if err != nil {
+			r.Log.Error(err, "Failed to create deployment:", "NamespaceName", nsName)
+			return ctrl.Result{}, err
+		}
+		r.Log.Info("Created deployment: ", "NamespaceName", nsName)
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if oxia.DeletionTimestamp != nil {
+		r.Log.Info("Deployment deletion is unsupported yet!")
+	}
+
+	// reconcile service
+	_service := &coreV1.Service{}
+	nsName = types.NamespacedName{Name: resourceName(Coordinator, oxia.Name), Namespace: oxia.Namespace}
+	err = r.Client.Get(ctx, nsName, _service)
+	if err != nil {
+		if !apiErrors.IsNotFound(err) {
+			r.Log.Error(err, "Failed to get service: ", "NamespaceName", nsName)
+			return ctrl.Result{}, err
+		}
+		r.Log.Info("Creating a new service:", "NamespaceName", nsName)
+		err = r.Client.Create(ctx, service(Coordinator, oxia, CoordinatorPorts))
```

Comment:
Is it possible to split the create and update of these resources(service, deployment, configmap,service account) into different packages to ensure that the current logic keeps simple?